terraform/modules/opensearch/main.tf

```diff
@@ -1,76 +1,138 @@
-resource "aws_iam_service_linked_role" "os" {
-  count            = var.create_os_service_role ? 1 : 0
-  aws_service_name = "es.amazonaws.com"
-}
-
-resource "aws_opensearch_domain" "os" {
-  domain_name    = local.domain_name
-  engine_version = var.opensearch_version
+resource "aws_opensearch_domain" "this" {
+  domain_name     = var.domain_name
```

Comment:
this would be where resource_prefix would be used

Reply:
- Updated as previously noted.